### PR TITLE
Inductor qlinear int8_fp32 with bmm

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #116604
* __->__ #116599
* #116937

**Summary**
Fix issue: https://github.com/pytorch/pytorch/issues/116492, `linear` will be decomposed into `bmm` when input dim exceeds 2 and not contiguous. Fix this issue by convert the pattern back into `qlinear`. This PR focus on int8_fp32 case, will follow up int8_bf16 case in next PR.

**Test Plan**
```
python -u -m pytest -s -v test_mkldnn_pattern_matcher.py -k test_qlinear_input_dim_exceeds_2_and_not_contiguous
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Comment by leslie-fang-intel** [comment]
cc @jiayisunx @jgong5 please kindly help to take a look of this PR.

File: torch/_inductor/fx_passes/quantization.py
Comment (jgong5) [inline torch/_inductor/fx_passes/quantization.py:1403]: add comment for the pattern here just like other paths?

**Review by jgong5 — APPROVED** [review]

File: torch/_inductor/fx_passes/quantization.py
Comment (leslie-fang-intel) [reply torch/_inductor/fx_passes/quantization.py:1403]: Thanks and add the comments for the linear patterns to do weight prepack.

**Comment by leslie-fang-intel** [comment]
Hi @atalman, this PR failed to land. looks like we still need all files in PR to match one merge rule?

**Comment by atalman** [comment]
@leslie-fang-intel and @jgong5 can we merge both groups ``x86 CPU quantization`` and ``CPU inductor`` these looks like closely related ?

**Comment by leslie-fang-intel** [comment]
> @leslie-fang-intel and @jgong5 can we merge both groups x86 CPU quantization and CPU inductor these looks like closely related ?

I am ok with it. The only concern is that some of the files under `x86 CPU quantization` like `aten/src/ATen/native/quantized/cpu/**`, `test/quantization/core/test_quantized_op.py` are inductor irreverent.

Another way just comes to my mind is maybe we can move `torch/_inductor/fx_passes/quantization.py` from `x86 CPU quantization` into `CPU inductor`, it should cover most of the PRs.

- Since in general, we will change `torch/_inductor/fx_passes/quantization.py` and `test/inductor/test_mkldnn_pattern_matcher.py` together in one Inductor lowering specific PR.
- The remaining files under `x86 CPU quantization` label are op and recipe specific and in general cases we can put them in separate PRs.

@jgong5 what do you think about it?

**Comment by jgong5** [comment]
> @jgong5 what do you think about it?

I'm open to merge these two modules into one in the merge rule.

**Comment by atalman** [comment]
@leslie-fang-intel and @jgong5 Sounds good, please post a PR to merge these rules. And tag me for approval. This should unblock this PR

**Comment by leslie-fang-intel** [comment]
> groups `x86 CPU quantization` and `CPU inductor`

Thanks @atalman, https://github.com/pytorch/pytorch/pull/116937 opened for it.

**Comment by leslie-fang-intel** [comment]
> @leslie-fang-intel and @jgong5 Sounds good, please post a PR to merge these rules. And tag me for approval. This should unblock this PR

Thanks @atalman, https://github.com/pytorch/pytorch/pull/116937 opened for it.

**Comment by malfet** [comment]
Hmm, why this PR depends on https://github.com/pytorch/pytorch/pull/116937 ? (They are completely unrelated, aren't they?)

**Comment by leslie-fang-intel** [comment]
> Hmm, why this PR depends on #116937 ? (They are completely unrelated, aren't they?)

Hi @malfet, this PR has changed 2 files which cross 2 different merge groups as `x86 CPU quantization` and `CPU inductor`. Per the discussion with @atalman above, due to the limitation of the current merging rules that prevent cross-checking all files among different merge groups, we can't merge it. So, we open [#116937](https://github.com/pytorch/pytorch/pull/116937) to workaround this issue per the discussion above.

**Pull request closed without merging**
